cardano-testnet/test/Spec/Plutus/Direct/TxInLockingPlutus.hs

```diff
@@ -0,0 +1,238 @@
+import           Control.Monad
+import           Data.Aeson (FromJSON(..), Value, (.:))
+import           Data.Bool (not)
+import           Data.Eq
+import           Data.Function
+import           Data.Functor ((<&>))
+import           Data.HashMap.Lazy (HashMap)
+import           Data.Int
+import           Data.List ((!!))
+import           Data.Maybe
+import           Data.Monoid (Last(..), (<>))
+import           Data.Text (Text)
+import           GHC.Real
+import           GHC.Num
+import           Hedgehog (Property, (===))
+import           Prelude (head)
+import           System.FilePath ((</>))
+import           Text.Show (Show(..))
+
+import qualified Data.Aeson as J
+import qualified Data.List as L
+import qualified Data.HashMap.Lazy as HM
+import qualified Data.Text as T
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+import qualified Hedgehog.Extras.Test.Base as H
+import qualified Hedgehog.Extras.Test.Concurrent as H
+import qualified Hedgehog.Extras.Test.File as H
+import qualified Hedgehog.Extras.Test.Process as H
+import qualified System.Directory as IO
+import qualified Test.Base as H
+import qualified Test.Process as H
+import qualified Testnet.Cardano as H
+import qualified Testnet.Conf as H
+
+{- HLINT ignore "Redundant <&>" -}
+{- HLINT ignore "Redundant return" -}
+{- HLINT ignore "Use let" -}
+
+data Utxo = Utxo
+  { address :: Text
+  , value :: HashMap Text Integer
+  } deriving (Eq, Show)
+
+instance FromJSON Utxo where
+  parseJSON = J.withObject "Coord" $ \v -> Utxo
+    <$> v .: "address"
+    <*> v .: "value"
+
+hprop_plutus :: Property
+hprop_plutus = H.integration . H.runFinallies . H.workspace "chairman" $ \tempAbsBasePath' -> do
+  projectBase <- H.note =<< H.noteIO . IO.canonicalizePath =<< H.getProjectBase
+  conf@H.Conf { H.tempBaseAbsPath, H.tempAbsPath } <- H.noteShowM $ H.mkConf tempAbsBasePath' Nothing
+
+  H.TestnetRuntime { H.bftSprockets, H.testnetMagic } <- H.testnet H.defaultTestnetOptions conf
+
+  -- path <- H.noteIO $ fromMaybe "" <$> IO.lookupEnv "PATH"
+
+  execConfig <- H.noteShow H.ExecConfig
+        { H.execConfigEnv = Last $ Just
+          [ ("CARDANO_NODE_SOCKET_PATH", IO.sprocketArgumentName (head bftSprockets))
+          ]
+        , H.execConfigCwd = Last $ Just tempBaseAbsPath
+        }
+
+  base <- H.note projectBase
+  work <- H.note tempAbsPath
+  utxoVKeyFile <- H.note $ tempAbsPath </> "shelley/utxo-keys/utxo1.vkey"
+  utxoSKeyFile <- H.note $ tempAbsPath </> "shelley/utxo-keys/utxo1.skey"
+
+  -- ("CARDANO_NODE_SOCKET_PATH", IO.sprocketArgumentName (head bftSprockets))
+
+  plutusScriptFileInUse <- H.note $ base </> "scripts/plutus/scripts/untyped-always-succeeds-txin.plutus"
+
+  -- This datum hash is the hash of the untyped 42
+  scriptDatumHash <- pure "9e1199a988ba72ffd6e9c269cadb3b53b5f360ff99f112d9b2ee30c4d74ad88b"
+  plutusRequiredSpace <- pure @_ @Integer 70000000
+  plutusRequiredTime <- pure @_ @Integer 70000000
```

Comment:
Strange, why the two `@`s?

Reply:
`pure` takes two type parameters.  The first is the `Applicative` type and the second is the value type.

Reply:
I simplified this.